File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int n= nums.size();
        unordered_map<int,int> numToFreq;
        for(auto num : nums){
            numToFreq[num]++;
        }
        vector<int>ans;
        for(auto itr : numToFreq){
            if(itr.second > (n/3) ) ans.emplace_back(itr.first);
        }
        return ans;
    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.py
class Solution:
    def majorityElement(self, nums):
        n = len(nums)
        numToFreq = dict()
        for num in nums:
            numToFreq[num] = numToFreq.get(num, 0) + 1
        
        ans = list()
        for num, freq in numToFreq.items():
            if freq > n // 3:
                ans.append(num)
        
        return ans
